src/devices/Vcnl4040/README.md

```diff
@@ -0,0 +1,111 @@
+  * **Ambient light measuring**: It measures ambient light levels in its vicinity. It provides readings that correspond to the illuminance (light intensity). The **measuring range is 0.0125 to 6553 lux**.
+  * **White light**: It has a white channel that can be used to capture visible white light. This channel can be valuable in applications where distinguishing between different colors or precise analysis of the visible light spectrum is required.
+  * **Configurable**: It is configurable with application specific settings, such as proximity and ambient light measurement resolution, noise cancellation and power consumption.
+  * **Interrupts**: It can be configured to trigger interrupts based on user-defined proximity or illuminance thresholds. Interrupts are signaled through the INT-pin of the device and by flags queryable by the host software.
+  * **I2C interface**: It is connected to a host (like RPi) with an I2C interface.
+
+## Official manufacturer documentation
+* [Datasheet](https://www.vishay.com/docs/84274/vcnl4040.pdf)
+* [Application note](https://www.vishay.com/docs/84307/designingvcnl4040.pdf)
+
+## Additional information and hints
+For a comprehensive description of the VCNL4040, please refer to the datasheet and, in particular, the application note. The following contains only some additional information about features and behaviors that may not be immediately apparent.
+
+Consider also the documentation of the API classes for further information.
+* Information on the configuration of the ambient light sensor can also be found in the documentation of the ```AmbientLightSensor``` and ```AmbientLightInterruptConfiguration``` classes.
+* Information on the configuration of the proximity sensor can also be found in the documentation of the ```ProximitySensor```, ```EmitterConfiguration```, ```ReceiverConfiguration``` and ```ProximityInterruptConfiguration``` classes.
+
+
+### Ambient light sensor
+#### Interdepence of parameters / settings:
+* **Integration Time, Resolution, and Range:**
+   - The integration time setting indirectly affects the resolution and range.
+   - Changing the integration time lead to adjustments in resolution and range.
+
+* **Threshold Configuration:**
+   - Configuring threshold levels involves considering the current resolution for accurate threshold values.
+   - Thresholds are set in counts, which are influenced by the current resolution setting.
+
+The illuminance measurement is configured by selecting an appropriate integration time. A shorter integration time increases the measurement range but reduces resolution. Moreover, the integration time determines the minimum cycle time for consecutive measurements. The table below illustrates the relationship between integration time, resolution, and measurement range.
+
+|Integration time [ms]|Resolution [lux]|Detection range [lux]|
+|-|-|-|
+|80|0.1|6335.5|
+|160|0.05|3276.7|
+|320|0.025|1638.3|
+|640|0.0125|819.1|
+
+#### Interrupts
```

Comment:
overall, can you add couple of lines of code to illustrate how to properly use each of those features?

Reply:
Well, most people will look at the few lines of code with the explanation to understand. So maybe just the simple example you have added to show that indeed it's simple for basic usage.

Reply:
Ok, agreed. I'll add an example section to the README.md.